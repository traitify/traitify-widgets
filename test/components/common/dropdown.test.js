/** @jest-environment jsdom */
import {act} from "react-test-renderer";
import Component from "components/common/dropdown";
import ComponentHandler from "support/component-handler";
import useGlobalMock from "support/hooks/use-global-mock";

describe("Dropdown", () => {
  let component;
  let options;
  let props;
  let queryElement;

  useGlobalMock(document, "querySelector");

  beforeEach(() => {
    options = [
      {name: "First", value: "one"},
      {name: "Second", value: "two"},
      {name: "Third", value: "three"}
    ];
    props = {
      id: "record-name",
      name: "name",
      onChange: jest.fn().mockName("onChange"),
      options
    };
    queryElement = {
      firstChild: {focus: jest.fn().mockName("focus")},
      focus: jest.fn().mockName("focus"),
      lastChild: {focus: jest.fn().mockName("focus")}
    };

    document.querySelector.mockReturnValue(queryElement);
  });

  describe("fancy", () => {
    beforeEach(() => {
      props = {...props, currentText: "Current", searchText: "Search"};
    });

    describe("keys", () => {
      let firstOption;
      let lastOption;
      let middleOption;

      beforeEach(() => {
        firstOption = props.options[0];
        lastOption = props.options[props.options.length - 1];
        middleOption = props.options[1];

        component = ComponentHandler.render(Component, {props});
        act(() => { component.findByText(props.searchText).props.onClick(); });
      });

      it("arrows up from first result", () => {
        const currentTarget = {nextSibling: {focus: jest.fn().mockName("focus")}};

        act(() => {
          component.findByText(firstOption.name)
            .props.onKeyDown({currentTarget, key: "ArrowUp", preventDefault: () => {}});
        });

        expect(currentTarget.nextSibling.focus).not.toHaveBeenCalled();
        expect(queryElement.firstChild.focus).not.toHaveBeenCalled();
        expect(queryElement.lastChild.focus).toHaveBeenCalled();
      });

      it("arrows up from last result", () => {
        const currentTarget = {previousSibling: {focus: jest.fn().mockName("focus")}};

        act(() => {
          component.findByText(lastOption.name)
            .props.onKeyDown({currentTarget, key: "ArrowUp", preventDefault: () => {}});
        });

        expect(currentTarget.previousSibling.focus).toHaveBeenCalled();
        expect(queryElement.firstChild.focus).not.toHaveBeenCalled();
        expect(queryElement.lastChild.focus).not.toHaveBeenCalled();
      });

      it("arrows up from search", () => {
        act(() => {
          component.instance.findByProps({placeholder: props.searchText})
            .props.onKeyDown({key: "ArrowUp", preventDefault: () => {}});
        });

        expect(queryElement.firstChild.focus).not.toHaveBeenCalled();
        expect(queryElement.lastChild.focus).toHaveBeenCalled();
      });

      it("arrows down from first result", () => {
        const currentTarget = {nextSibling: {focus: jest.fn().mockName("focus")}};

        act(() => {
          component.findByText(firstOption.name)
            .props.onKeyDown({currentTarget, key: "ArrowDown", preventDefault: () => {}});
        });

        expect(currentTarget.nextSibling.focus).toHaveBeenCalled();
        expect(queryElement.firstChild.focus).not.toHaveBeenCalled();
        expect(queryElement.lastChild.focus).not.toHaveBeenCalled();
      });

      it("arrows down from last result", () => {
        const currentTarget = {previousSibling: {focus: jest.fn().mockName("focus")}};

        act(() => {
          component.findByText(lastOption.name)
            .props.onKeyDown({currentTarget, key: "ArrowDown", preventDefault: () => {}});
        });

        expect(currentTarget.previousSibling.focus).not.toHaveBeenCalled();
        expect(queryElement.firstChild.focus).toHaveBeenCalled();
        expect(queryElement.lastChild.focus).not.toHaveBeenCalled();
      });

      it("arrows down from search", () => {
        act(() => {
          component.instance.findByProps({placeholder: props.searchText})
            .props.onKeyDown({key: "ArrowDown", preventDefault: () => {}});
        });

        expect(queryElement.firstChild.focus).toHaveBeenCalled();
        expect(queryElement.lastChild.focus).not.toHaveBeenCalled();
      });

      it("selects with enter", () => {
        const currentTarget = {
          click: jest.fn().mockName("click"),
          nextSibling: {focus: jest.fn().mockName("focus")},
          previousSibling: {focus: jest.fn().mockName("focus")}
        };

        act(() => {
          component.findByText(middleOption.name)
            .props.onKeyDown({currentTarget, key: "Enter", preventDefault: () => {}});
        });

        expect(currentTarget.click).toHaveBeenCalled();
        expect(currentTarget.nextSibling.focus).not.toHaveBeenCalled();
        expect(currentTarget.previousSibling.focus).not.toHaveBeenCalled();
        expect(queryElement.firstChild.focus).not.toHaveBeenCalled();
        expect(queryElement.lastChild.focus).not.toHaveBeenCalled();
      });

      it("selects with space", () => {
        const currentTarget = {
          click: jest.fn().mockName("click"),
          nextSibling: {focus: jest.fn().mockName("focus")},
          previousSibling: {focus: jest.fn().mockName("focus")}
        };

        act(() => {
          component.findByText(middleOption.name)
            .props.onKeyDown({currentTarget, code: "Space", preventDefault: () => {}});
        });

        expect(currentTarget.click).toHaveBeenCalled();
        expect(currentTarget.nextSibling.focus).not.toHaveBeenCalled();
        expect(currentTarget.previousSibling.focus).not.toHaveBeenCalled();
        expect(queryElement.firstChild.focus).not.toHaveBeenCalled();
        expect(queryElement.lastChild.focus).not.toHaveBeenCalled();
      });
    });

    describe("search", () => {
      beforeEach(() => {
        const currentOption = props.options[2];
        props.value = currentOption.value;
        component = ComponentHandler.render(Component, {props});
        act(() => { component.findByText(currentOption.name).parent.parent.props.onClick(); });
      });

      it("searches by name", () => {
        act(() => {
          component.instance.findByProps({placeholder: props.searchText})
            .props.onChange({target: {value: "Sec"}});
        });

        expect(component.tree).toMatchSnapshot();
      });

      it("searches by value", () => {
        act(() => {
          component.instance.findByProps({placeholder: props.searchText})
            .props.onChange({target: {value: "Tw"}});
        });

        expect(component.tree).toMatchSnapshot();
      });

      it("searches without results", () => {
        act(() => {
          component.instance.findByProps({placeholder: props.searchText})
            .props.onChange({target: {value: "Twenty"}});
        });

        expect(component.tree).toMatchSnapshot();

        act(() => { component.findByText(props.searchText).props.onClick(); });

        expect(component.tree).toMatchSnapshot();
      });
    });

    it("renders component", () => {
      component = ComponentHandler.render(Component, {props});

      expect(component.tree).toMatchSnapshot();
    });

    it("renders component with options", () => {
      component = ComponentHandler.render(Component, {props});
      act(() => { component.findByText(props.searchText).props.onClick(); });

      expect(component.tree).toMatchSnapshot();
    });

    it("renders component with value", () => {
      props.value = props.options[2].value;
      component = ComponentHandler.render(Component, {props});

      expect(component.tree).toMatchSnapshot();
    });

    describe("select", () => {
      it("selects current option", () => {
        const option = props.options[2];
        props.value = option.value;
        component = ComponentHandler.render(Component, {props});
        act(() => { component.findByText(option.name).parent.parent.props.onClick(); });
        act(() => {
          component.instance.find((element) => (
            element.type === "button"
            && element.children.includes(option.name)
            && element.children.includes(props.currentText)
          )).props.onClick();
        });

        expect(component.tree).toMatchSnapshot();
      });

      it("selects option", () => {
        const currentOption = props.options[2];
        const option = props.options[1];
        props.value = currentOption.value;
        component = ComponentHandler.render(Component, {props});
        act(() => { component.findByText(currentOption.name).parent.parent.props.onClick(); });
        act(() => {
          component.instance.find((element) => (
            element.type === "button"
            && element.children.includes(option.name)
          )).props.onClick();
        });

        expect(component.tree).toMatchSnapshot();
      });
    });
  });

  describe("options", () => {
    it("renders with objects", () => {
      component = ComponentHandler.render(Component, {props});

      expect(component.tree).toMatchSnapshot();
    });

    it("renders with strings", () => {
      props.options = ["one", "two", "three"];
      component = ComponentHandler.render(Component, {props});

      expect(component.tree).toMatchSnapshot();
    });
  });

  describe("simple", () => {
    it("renders component", () => {
      component = ComponentHandler.render(Component, {props});

      expect(component.tree).toMatchSnapshot();
    });
  });
});
