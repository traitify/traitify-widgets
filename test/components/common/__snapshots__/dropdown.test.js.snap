// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown fancy renders component 1`] = `
<label
  className="container"
  htmlFor="record-name"
>
  <button
    className="current"
    onClick={[Function]}
    type="button"
  >
    Search
    <span
      className="traitify--icon-fa"
      dangerouslySetInnerHTML={
        {
          "__html": "<svg aria-hidden="true" focusable="false" data-prefix="fas" data-icon="angle-up" class="svg-inline--fa fa-angle-up" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><path fill="currentColor" d="M201.4 137.4c12.5-12.5 32.8-12.5 45.3 0l160 160c12.5 12.5 12.5 32.8 0 45.3s-32.8 12.5-45.3 0L224 205.3 86.6 342.6c-12.5 12.5-32.8 12.5-45.3 0s-12.5-32.8 0-45.3l160-160z"></path></svg>",
        }
      }
    />
  </button>
</label>
`;

exports[`Dropdown fancy renders component with options 1`] = `
<label
  className="container"
  htmlFor="record-name"
>
  <button
    className="current"
    onClick={[Function]}
    type="button"
  >
    Search
    <span
      className="traitify--icon-fa"
      dangerouslySetInnerHTML={
        {
          "__html": "<svg aria-hidden="true" focusable="false" data-prefix="fas" data-icon="angle-down" class="svg-inline--fa fa-angle-down" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><path fill="currentColor" d="M201.4 374.6c12.5 12.5 32.8 12.5 45.3 0l160-160c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L224 306.7 86.6 169.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3l160 160z"></path></svg>",
        }
      }
    />
  </button>
  <div
    className="dropdown"
  >
    <div
      className="search"
    >
      <span
        alt="Search"
        className="traitify--icon-fa"
        dangerouslySetInnerHTML={
          {
            "__html": "<svg aria-hidden="true" focusable="false" data-prefix="fas" data-icon="magnifying-glass" class="svg-inline--fa fa-magnifying-glass" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><path fill="currentColor" d="M416 208c0 45.9-14.9 88.3-40 122.7L502.6 457.4c12.5 12.5 12.5 32.8 0 45.3s-32.8 12.5-45.3 0L330.7 376c-34.4 25.2-76.8 40-122.7 40C93.1 416 0 322.9 0 208S93.1 0 208 0S416 93.1 416 208zM208 352a144 144 0 1 0 0-288 144 144 0 1 0 0 288z"></path></svg>",
          }
        }
      />
      <input
        autoComplete="off"
        id="record-name"
        name="search"
        onChange={[Function]}
        onKeyDown={[Function]}
        placeholder="Search"
        type="text"
        value=""
      />
    </div>
    <div
      className="options"
    >
      <button
        onClick={[Function]}
        onKeyDown={[Function]}
        tabIndex={0}
        type="button"
      >
        First
      </button>
      <button
        onClick={[Function]}
        onKeyDown={[Function]}
        tabIndex={0}
        type="button"
      >
        Second
      </button>
      <button
        onClick={[Function]}
        onKeyDown={[Function]}
        tabIndex={0}
        type="button"
      >
        Third
      </button>
    </div>
  </div>
</label>
`;

exports[`Dropdown fancy renders component with value 1`] = `
<label
  className="container"
  htmlFor="record-name"
>
  <button
    className="current"
    onClick={[Function]}
    type="button"
  >
    <div>
      <span
        className="collapse"
      >
        Third
      </span>
      <span
        className="stiff"
      >
         (
        Current
        )
      </span>
    </div>
    <span
      className="traitify--icon-fa"
      dangerouslySetInnerHTML={
        {
          "__html": "<svg aria-hidden="true" focusable="false" data-prefix="fas" data-icon="angle-up" class="svg-inline--fa fa-angle-up" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><path fill="currentColor" d="M201.4 137.4c12.5-12.5 32.8-12.5 45.3 0l160 160c12.5 12.5 12.5 32.8 0 45.3s-32.8 12.5-45.3 0L224 205.3 86.6 342.6c-12.5 12.5-32.8 12.5-45.3 0s-12.5-32.8 0-45.3l160-160z"></path></svg>",
        }
      }
    />
  </button>
</label>
`;

exports[`Dropdown fancy search searches by name 1`] = `
<label
  className="container"
  htmlFor="record-name"
>
  <button
    className="current"
    onClick={[Function]}
    type="button"
  >
    <div>
      <span
        className="collapse"
      >
        Third
      </span>
      <span
        className="stiff"
      >
         (
        Current
        )
      </span>
    </div>
    <span
      className="traitify--icon-fa"
      dangerouslySetInnerHTML={
        {
          "__html": "<svg aria-hidden="true" focusable="false" data-prefix="fas" data-icon="angle-down" class="svg-inline--fa fa-angle-down" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><path fill="currentColor" d="M201.4 374.6c12.5 12.5 32.8 12.5 45.3 0l160-160c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L224 306.7 86.6 169.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3l160 160z"></path></svg>",
        }
      }
    />
  </button>
  <div
    className="dropdown"
  >
    <div
      className="search"
    >
      <span
        alt="Search"
        className="traitify--icon-fa"
        dangerouslySetInnerHTML={
          {
            "__html": "<svg aria-hidden="true" focusable="false" data-prefix="fas" data-icon="magnifying-glass" class="svg-inline--fa fa-magnifying-glass" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><path fill="currentColor" d="M416 208c0 45.9-14.9 88.3-40 122.7L502.6 457.4c12.5 12.5 12.5 32.8 0 45.3s-32.8 12.5-45.3 0L330.7 376c-34.4 25.2-76.8 40-122.7 40C93.1 416 0 322.9 0 208S93.1 0 208 0S416 93.1 416 208zM208 352a144 144 0 1 0 0-288 144 144 0 1 0 0 288z"></path></svg>",
          }
        }
      />
      <input
        autoComplete="off"
        id="record-name"
        name="search"
        onChange={[Function]}
        onKeyDown={[Function]}
        placeholder="Search"
        type="text"
        value="Sec"
      />
    </div>
    <div
      className="options"
    >
      <button
        onClick={[Function]}
        onKeyDown={[Function]}
        tabIndex={0}
        type="button"
      >
        Second
      </button>
    </div>
  </div>
</label>
`;

exports[`Dropdown fancy search searches by value 1`] = `
<label
  className="container"
  htmlFor="record-name"
>
  <button
    className="current"
    onClick={[Function]}
    type="button"
  >
    <div>
      <span
        className="collapse"
      >
        Third
      </span>
      <span
        className="stiff"
      >
         (
        Current
        )
      </span>
    </div>
    <span
      className="traitify--icon-fa"
      dangerouslySetInnerHTML={
        {
          "__html": "<svg aria-hidden="true" focusable="false" data-prefix="fas" data-icon="angle-down" class="svg-inline--fa fa-angle-down" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><path fill="currentColor" d="M201.4 374.6c12.5 12.5 32.8 12.5 45.3 0l160-160c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L224 306.7 86.6 169.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3l160 160z"></path></svg>",
        }
      }
    />
  </button>
  <div
    className="dropdown"
  >
    <div
      className="search"
    >
      <span
        alt="Search"
        className="traitify--icon-fa"
        dangerouslySetInnerHTML={
          {
            "__html": "<svg aria-hidden="true" focusable="false" data-prefix="fas" data-icon="magnifying-glass" class="svg-inline--fa fa-magnifying-glass" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><path fill="currentColor" d="M416 208c0 45.9-14.9 88.3-40 122.7L502.6 457.4c12.5 12.5 12.5 32.8 0 45.3s-32.8 12.5-45.3 0L330.7 376c-34.4 25.2-76.8 40-122.7 40C93.1 416 0 322.9 0 208S93.1 0 208 0S416 93.1 416 208zM208 352a144 144 0 1 0 0-288 144 144 0 1 0 0 288z"></path></svg>",
          }
        }
      />
      <input
        autoComplete="off"
        id="record-name"
        name="search"
        onChange={[Function]}
        onKeyDown={[Function]}
        placeholder="Search"
        type="text"
        value="Tw"
      />
    </div>
    <div
      className="options"
    >
      <button
        onClick={[Function]}
        onKeyDown={[Function]}
        tabIndex={0}
        type="button"
      >
        Second
      </button>
    </div>
  </div>
</label>
`;

exports[`Dropdown fancy search searches without results 1`] = `
<label
  className="container"
  htmlFor="record-name"
>
  <button
    className="current"
    onClick={[Function]}
    type="button"
  >
    <div>
      <span
        className="collapse"
      >
        Third
      </span>
      <span
        className="stiff"
      >
         (
        Current
        )
      </span>
    </div>
    <span
      className="traitify--icon-fa"
      dangerouslySetInnerHTML={
        {
          "__html": "<svg aria-hidden="true" focusable="false" data-prefix="fas" data-icon="angle-down" class="svg-inline--fa fa-angle-down" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><path fill="currentColor" d="M201.4 374.6c12.5 12.5 32.8 12.5 45.3 0l160-160c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L224 306.7 86.6 169.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3l160 160z"></path></svg>",
        }
      }
    />
  </button>
  <div
    className="dropdown"
  >
    <div
      className="search"
    >
      <span
        alt="Search"
        className="traitify--icon-fa"
        dangerouslySetInnerHTML={
          {
            "__html": "<svg aria-hidden="true" focusable="false" data-prefix="fas" data-icon="magnifying-glass" class="svg-inline--fa fa-magnifying-glass" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><path fill="currentColor" d="M416 208c0 45.9-14.9 88.3-40 122.7L502.6 457.4c12.5 12.5 12.5 32.8 0 45.3s-32.8 12.5-45.3 0L330.7 376c-34.4 25.2-76.8 40-122.7 40C93.1 416 0 322.9 0 208S93.1 0 208 0S416 93.1 416 208zM208 352a144 144 0 1 0 0-288 144 144 0 1 0 0 288z"></path></svg>",
          }
        }
      />
      <input
        autoComplete="off"
        id="record-name"
        name="search"
        onChange={[Function]}
        onKeyDown={[Function]}
        placeholder="Search"
        type="text"
        value="Twenty"
      />
    </div>
    <div
      className="options"
    >
      <button
        onClick={[Function]}
        onKeyDown={[Function]}
        tabIndex={0}
        type="button"
      >
        Search
      </button>
    </div>
  </div>
</label>
`;

exports[`Dropdown fancy search searches without results 2`] = `
<label
  className="container"
  htmlFor="record-name"
>
  <button
    className="current"
    onClick={[Function]}
    type="button"
  >
    <div>
      <span
        className="collapse"
      >
        Third
      </span>
      <span
        className="stiff"
      >
         (
        Current
        )
      </span>
    </div>
    <span
      className="traitify--icon-fa"
      dangerouslySetInnerHTML={
        {
          "__html": "<svg aria-hidden="true" focusable="false" data-prefix="fas" data-icon="angle-down" class="svg-inline--fa fa-angle-down" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><path fill="currentColor" d="M201.4 374.6c12.5 12.5 32.8 12.5 45.3 0l160-160c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L224 306.7 86.6 169.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3l160 160z"></path></svg>",
        }
      }
    />
  </button>
  <div
    className="dropdown"
  >
    <div
      className="search"
    >
      <span
        alt="Search"
        className="traitify--icon-fa"
        dangerouslySetInnerHTML={
          {
            "__html": "<svg aria-hidden="true" focusable="false" data-prefix="fas" data-icon="magnifying-glass" class="svg-inline--fa fa-magnifying-glass" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><path fill="currentColor" d="M416 208c0 45.9-14.9 88.3-40 122.7L502.6 457.4c12.5 12.5 12.5 32.8 0 45.3s-32.8 12.5-45.3 0L330.7 376c-34.4 25.2-76.8 40-122.7 40C93.1 416 0 322.9 0 208S93.1 0 208 0S416 93.1 416 208zM208 352a144 144 0 1 0 0-288 144 144 0 1 0 0 288z"></path></svg>",
          }
        }
      />
      <input
        autoComplete="off"
        id="record-name"
        name="search"
        onChange={[Function]}
        onKeyDown={[Function]}
        placeholder="Search"
        type="text"
        value="Twenty"
      />
    </div>
    <div
      className="options"
    >
      <button
        onClick={[Function]}
        onKeyDown={[Function]}
        tabIndex={0}
        type="button"
      >
        Search
      </button>
    </div>
  </div>
</label>
`;

exports[`Dropdown fancy select selects current option 1`] = `
<label
  className="container"
  htmlFor="record-name"
>
  <button
    className="current"
    onClick={[Function]}
    type="button"
  >
    <div>
      <span
        className="collapse"
      >
        Third
      </span>
      <span
        className="stiff"
      >
         (
        Current
        )
      </span>
    </div>
    <span
      className="traitify--icon-fa"
      dangerouslySetInnerHTML={
        {
          "__html": "<svg aria-hidden="true" focusable="false" data-prefix="fas" data-icon="angle-up" class="svg-inline--fa fa-angle-up" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><path fill="currentColor" d="M201.4 137.4c12.5-12.5 32.8-12.5 45.3 0l160 160c12.5 12.5 12.5 32.8 0 45.3s-32.8 12.5-45.3 0L224 205.3 86.6 342.6c-12.5 12.5-32.8 12.5-45.3 0s-12.5-32.8 0-45.3l160-160z"></path></svg>",
        }
      }
    />
  </button>
</label>
`;

exports[`Dropdown fancy select selects option 1`] = `
<label
  className="container"
  htmlFor="record-name"
>
  <button
    className="current"
    onClick={[Function]}
    type="button"
  >
    <div>
      <span
        className="collapse"
      >
        Third
      </span>
      <span
        className="stiff"
      >
         (
        Current
        )
      </span>
    </div>
    <span
      className="traitify--icon-fa"
      dangerouslySetInnerHTML={
        {
          "__html": "<svg aria-hidden="true" focusable="false" data-prefix="fas" data-icon="angle-up" class="svg-inline--fa fa-angle-up" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><path fill="currentColor" d="M201.4 137.4c12.5-12.5 32.8-12.5 45.3 0l160 160c12.5 12.5 12.5 32.8 0 45.3s-32.8 12.5-45.3 0L224 205.3 86.6 342.6c-12.5 12.5-32.8 12.5-45.3 0s-12.5-32.8 0-45.3l160-160z"></path></svg>",
        }
      }
    />
  </button>
</label>
`;

exports[`Dropdown options renders with objects 1`] = `
<label
  className="container"
  htmlFor="record-name"
>
  <select
    id="record-name"
    name="name"
    onChange={[MockFunction onChange]}
  >
    <option
      value="one"
    >
      First
    </option>
    <option
      value="two"
    >
      Second
    </option>
    <option
      value="three"
    >
      Third
    </option>
  </select>
</label>
`;

exports[`Dropdown options renders with strings 1`] = `
<label
  className="container"
  htmlFor="record-name"
>
  <select
    id="record-name"
    name="name"
    onChange={[MockFunction onChange]}
  >
    <option
      value="one"
    >
      one
    </option>
    <option
      value="two"
    >
      two
    </option>
    <option
      value="three"
    >
      three
    </option>
  </select>
</label>
`;

exports[`Dropdown simple renders component 1`] = `
<label
  className="container"
  htmlFor="record-name"
>
  <select
    id="record-name"
    name="name"
    onChange={[MockFunction onChange]}
  >
    <option
      value="one"
    >
      First
    </option>
    <option
      value="two"
    >
      Second
    </option>
    <option
      value="three"
    >
      Third
    </option>
  </select>
</label>
`;
